test/e2e/storage/testsuites/base.go

```diff
@@ -421,6 +427,64 @@ func convertTestConfig(in *PerTestConfig) volume.TestConfig {
 	}
 }
 
+// getSizeRangesIntersection takes two instances of storage size ranges and determines the
+// intersection of the intervals (if it exists) and return the minimum of the intersection
+// to be used as the claim size for the test.
+// if value not set, that means there's no minimum or maximum size limitation and we set default size for it.
```

Comment:
I think this logic of setting the default size before the comparison is wrong.

Consider this example: `first` is empty (no min, no max set) and `second` is `Min = 1Mi`, `Max = 1Mi`. My expectation is that this results in `1Mi` being returned. I think what the code does instead is replace `first.Min` with `defaultMinSize  = "5Gi"` and then fails to find an intersection.

The bottom line is, you need a table-driven unit test for this function. It's non-trivial to ensure that the corner cases are handled.

Reply:
I have updated this PR base on all points you mentioned. There's a UT  Test_getSizeRangesIntersection to test all corner cases. 
Please tell me if I miss something.